### PR TITLE
refact: refactor dep detection for webpack v5 BUT src/** missing

### LINKED ISSUE #20: Webpack 5 support - the fileDependencies always empty
image

### PR DESCRIPTION
**Pull request by deleonio**
It relates to the following issue:
* Fixes #20

### CODE REVIEW

**Comment by deleonio** [comment]
@sokra can you give us a few pointers.

**Comment by deleonio** [comment]
@rpokorny the tests not working in orginial master and forked branch too.

File: index.js [hunk 1]
```diff
@@ -40,8 +40,15 @@ module.exports = class WebpackCopyModulesPlugin {
...
         fileDependencies = new Set();
 
-    compilation.modules.forEach(module => (module.buildInfo.fileDependencies ||[])
-        .forEach(fileDependencies.add.bind(fileDependencies)));
+    compilation.modules.forEach(module => {
+      if (typeof module === 'object' && module !== null &&
+        typeof module.buildInfo === 'object' && module.buildInfo !== null &&
+        typeof module.buildInfo.snapshot === 'object' && module.buildInfo.snapshot !== null
+      ) {
+        (module.buildInfo.snapshot.managedFiles ||[])
```
Comment (sokra) [inline index.js:48]: You probably want `snapshot.getFileIterable()`.

**Comment by rpokorny** [comment]
> @rpokorny the tests not working in orginial master and forked branch too.

Hello.  Could you clarify what you mean by this? I just ran the tests on master today and they are passing for me locally.

**Comment by rpokorny** [comment]
The devDependency on webpack sets the version that is tested against.  This PR does not change that, so the tests are still running against webpack 4.  I believe you'd want to run them against webpack 5?

**Comment by rpokorny** [comment]
I've created an [alternative PR](https://github.com/sonatype-nexus-community/copy-modules-webpack-plugin/pull/22) which I believe addresses this more thoroughly.

**Comment by deleonio** [comment]
Thank you, it works.

**Pull request closed without merging**